crates/services/relayer/src/ports/tests.rs

```diff
@@ -69,13 +69,13 @@ fn insert_always_raises_da_height_monotonically() {
     db.insert_messages(&5u64.into(), &messages[..5]).unwrap();
 }
 
-#[test_case(None, 0, 0)]
-#[test_case(None, 10, 10)]
-#[test_case(0, 10, 10)]
-#[test_case(0, None, 0)]
-#[test_case(10, 11, 11)]
-#[test_case(11, None, 11)]
-#[test_case(11, None, 10)]
+#[test_case(None, 0, 0; "can set DA height to 0 when there is none available")]
+#[test_case(None, 10, 10; "can set DA height to 10 when there is none available")]
+#[test_case(0, 10, 10; "can set DA height to 10 when it is 0")]
+#[test_case(0, None, 0; "inserts are bypassed when height goes from 0 to 0")]
+#[test_case(10, 11, 11; "can set DA height to 11 when it is 10")]
+#[test_case(11, None, 11; "inserts are bypassed when height goes from 11 to 11")]
+#[test_case(11, None, 10; "inserts are bypassed when height reverted from 11 to 10")]
 fn set_raises_da_height_monotonically(
```

Comment:
These tests are confusing. Especially since there are no checks on output.

I could rewrite `set_finalized_da_height_to_at_least` to do nothing and it would still pass all of these tests:
```
fn set_finalized_da_height_to_at_least(
        &mut self,
        height: &DaBlockHeight,
    ) -> StorageResult<()> {
        let mut db_tx = self.transaction();
        let db = db_tx.as_mut();
        let _ = db.get(&METADATA_KEY);
        db.insert(&METADATA_KEY, 0)?;
        Ok(())
    }
```
:(

It would be nice if these tests checked storage after to see if the values are correct. It would be nice to have some failing tests! 

`mockall` is convenient, but it often puts the tests' focus on internal details. Tests should check how the interface behaves as a black box. 
`test_case` helps keep tests DRY, but it pushes tests toward reuse rather that coherence IMO. 

"Given, When, Then" would be useful here.